### DISCUSSION TITLE
How to fix icon in contexmenu in material design in XAML toolkit?

### DISCUSSION THREAD

**Discussion by VahidEra** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3072] [Q&A] [answered]
Hi guys,
I have a contexmenu in my WPF project using material design and I want to add an icon to one of the items but when I add icon to it my icon not showing properly

***This is my contexmenu with adding icon:***
![Screenshot (2)](https://user-images.githubusercontent.com/27216710/216549927-bb18ea43-893c-45c9-98e0-4b084852dbd3.png)

***and this is my icon:***
![2023-02-03_114635](https://user-images.githubusercontent.com/27216710/216550292-26f8c610-534f-4924-8402-f6888bfbf427.png)

***This is cod:***
`            <MenuItem
                x:Name="itmAddToPlayList" 
                Header="Add to playlist"
                Height="60"
                FontFamily ="ROBOTO" 
                FontWeight="Normal"
                FontSize="13"
                Icon="{materialDesign:PackIcon Kind=PlaylistMusicOutline, Size=30 }"/>`

                
>Where is my mistake?

**Comment by Keboo** [comment, marked as answer]
At present the reason it is being clipped is because in the MenuItem template the icon container [is declared with a fixed Width and Height of 16](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/blob/master/MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.Menu.xaml#L100-L107) like this:
```xaml
<ContentPresenter x:Name="Icon"
                  Width="16"
                  Height="16"
                  HorizontalAlignment="Left"
                  VerticalAlignment="Center"
                  Content="{TemplateBinding Icon}"
                  ContentSource="Icon"
                  SnapsToDevicePixels="{TemplateBinding SnapsToDevicePixels}" />
```
So your icon with a size of 30 is being clipped. That fixed sizing is probably in correct and should really be changed so that when the menu height is increased (like in your example) the is appropriately sized.

**Reply by VahidEra** [reply]
@Keboo dear kevin, Thank you and I understand. I am very grateful for your guidance, dear friend.
